### ISSUE TITLE
Linting with 0.73.0 doesn't accept property `TargetRole` of `AWS::RDS::DBProxyEndpoint`

### ISSUE THREAD

**Issue by matthiasgubler** [aws-cloudformation/cfn-lint#2579]
### CloudFormation Lint Version

0.73.0

### What operating system are you using?

Mac

### Describe the bug

When performing the `cfn-lint` I get the following error:
```
E3002 Invalid Property Resources/RDSProxyReadOnlyEndpoint/Properties/TargetRole
```

This property still appears on the doc here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbproxyendpoint.html

### Expected behavior

No linting issues

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Resources:
  RDSProxyReadOnlyEndpoint:
    Type: AWS::RDS::DBProxyEndpoint
    Properties:
      DBProxyEndpointName: "rds-proxy-RO-endpoint"
      DBProxyName: 'name'
      VpcSubnetIds:
        - 123
      TargetRole: READ_ONLY

```

**Comment by matthiasgubler** [comment]
Relates to: https://github.com/aws/aws-cdk/pull/23984/files#diff-448833823999f11d01885d768d952386baf95b8d41a7096616d5d96e81acce0d

See also a similar issue in `aws-cdk` repo: https://github.com/aws/aws-cdk/issues/24092

**Comment by kddejong** [comment]
`TargetRole` is listed as a `readOnlyProperties` in its [schema](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-type-schemas.html).  Since its under readOnlyProperties its being stripped out.

**Comment by matthiasgubler** [comment]
But why is it so? It's an input property that can also be set via the UI?
Is there an issue in the schema?
<img width="569" alt="image" src="https://user-images.githubusercontent.com/2006388/218199503-547398a6-0d55-43a9-a5c6-8ba6d9608025.png">

**Comment by kddejong** [comment]
yes and no.  I can patch in that property but readOnlyProperties are used for GetAtts.  Sometimes teams want to use these properties for inputs and also supply them as GetAtts.
